### PR TITLE
LTI Integration migration and schema

### PR DESCRIPTION
**Pull request by nicklathe**
- Add LTI Integration migration
  - Add LTI Integration schema changes

This is part 1 of a 2 part PR adding a new LTI Integration model.

This PR adds only the migration and schema, per the standard for adding new Models. The [second PR](https://github.com/code-dot-org/code-dot-org/pull/51734) will follow this, and will add the other related files to this new Model.

## Links

- [Second dependent PR](https://github.com/code-dot-org/code-dot-org/pull/51734)

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Comment by sureshc** [comment]
Do we need the ability to logically ('soft') delete. These records using the `paranoia` gem and a `deleted_at` column? I haven't read our LTI requirements / specs. I assume the number of LTI records we expect to save is relatively low (less than 10K?). One per school district basically?

**Comment by nicklathe** [comment]
> Do we need the ability to logically ('soft') delete. These records using the `paranoia` gem and a `deleted_at` column? I haven't read our LTI requirements / specs. I assume the number of LTI records we expect to save is relatively low (less than 10K?). One per school district basically?

I don't see a reason for us needing to. We may not even offer a way for a self-service delete of an integration, as deleting the LTI Tool from the LMS will effectively disable the integration.

However, if we did want to offer this, is it as easy as adding a `t.deleted_at` column?

**Comment by nicklathe** [comment]
And yes, it _should_ be about 1 per school district under normal circumstances.

File: dashboard/db/migrate/20230503220558_create_lti_integrations.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+class CreateLtiIntegrations < ActiveRecord::Migration[6.0]
+  def change
```
Comment (elf-code) [inline dashboard/db/migrate/20230503220558_create_lti_integrations.rb:2]: I think for the soft deletion `paranoia` gem, you would just add `acts_as_paranoid` here on line 2 (like in other models - https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/channel_token.rb#L22 )

I think it's ok either way, but probably easier to start w/ the soft-deletion pattern now.

Comment (nicklathe) [reply dashboard/db/migrate/20230503220558_create_lti_integrations.rb:2]: That would be added to the Model, not this migration though (looking at the linked file). Right? Doing a global search, I don't see this being added to a migration anywhere.

Comment (nicklathe) [reply dashboard/db/migrate/20230503220558_create_lti_integrations.rb:2]: I agree, seems like we should start with the soft-deletion pattern. I'll add it to the related PR that adds the lti_integrations model.

Comment (sureshc) [reply dashboard/db/migrate/20230503220558_create_lti_integrations.rb:2]: Given the small size of the dataset stored in this table, we can easily add a `deleted_at` column in the future if we want to support soft deletion.

**Review by sureshc — APPROVED** [review]

**Pull request merged**